third_party/nvfuser/test/test_gpu_expr_simplifier.cpp

```diff
@@ -164,6 +164,11 @@ TEST_F(NVFuserTest, FusionEliminateTrivialComputation_CUDA) {
   // a && false -> false
   TORCH_CHECK(simplifyExpr(IrBuilder::andExpr(b, f))->sameAs(f));
 
+  // a && a -> a
```

Comment:
nit: This isn't strictly correct in C++ as `a` could have side effects. As far as I remember, we do not generate predicates with side effects, so this should be fine.

Reply:
🤯🤯🤯🤯🤯 You are right. I will leave a comment in the code.